lib/records/P_RecCore.cc

```diff
@@ -377,22 +377,25 @@ RecSetRecord(RecT rec_type, const char *name, RecDataT data_type, RecData *data,
       if ((data_type != RECD_NULL) && (r1->data_type != data_type)) {
         err = REC_ERR_FAIL;
       } else {
+        bool rec_set = false;
```

Comment:
Nit pick, but what about "rec_was_updated" or "rec_updated_p"? I think it's clearer.